avocado/utils/astring.py

```diff
@@ -219,7 +219,7 @@ def string_safe_encode(string):
     try:
         return string.encode("utf-8")
     except UnicodeDecodeError:
-        return string.decode("utf-8").encode("utf-8")
+        return string.decode("utf-8", "ignore").encode("utf-8")
```

Comment:
Isn't there something weird here? I mean, the statement under `try/catch` is an **encode** operation, which, according to the documentation:

```
Help on built-in function encode:

encode(...)
    S.encode([encoding[,errors]]) -> object
    
    Encodes S using the codec registered for encoding. encoding defaults
    to the default encoding. errors may be given to set a different error
    handling scheme. Default is 'strict' meaning that encoding errors raise
    a UnicodeEncodeError. Other possible values are 'ignore', 'replace' and
    'xmlcharrefreplace' as well as any other name registered with
    codecs.register_error that is able to handle UnicodeEncodeErrors.
```

Raises Unicode**Encode**Errors, and not Unicode**Decode**Errors.